python/ray/train/tensorflow/tensorflow_checkpoint.py

```diff
@@ -48,9 +73,175 @@ def from_model(
         checkpoint = cls.from_dict(
             {PREPROCESSOR_KEY: preprocessor, MODEL_KEY: model.get_weights()}
         )
+        checkpoint._flavor = cls.Flavor.MODEL_WEIGHTS
+        return checkpoint
+
+    @classmethod
+    def from_h5(
+        cls, file_path: str, *, preprocessor: Optional["Preprocessor"] = None
+    ) -> "TensorflowCheckpoint":
+        """Create a :py:class:`~ray.air.checkpoint.Checkpoint` that stores a Keras
+        model - from SavedModel format.
+
+        The path must maintain valid even after this function returns.
+
+        Args:
+            file_path: The same path as used by ``model.save(path)``.
+            preprocessor: A fitted preprocessor to be applied before inference.
+
+        Returns:
+            A :py:class:`TensorflowCheckpoint` converted from h5 format.
+
+        Examples:
+
+            >>> import tensorflow as tf
+
+            >>> import ray
+            >>> from ray.train.batch_predictor import BatchPredictor
+            >>> from ray.train.tensorflow import (
+            ...     TensorflowCheckpoint, TensorflowTrainer, TensorflowPredictor
+            ... )
+            >>> from ray.air import session
+            >>> from ray.air.config import ScalingConfig
+
+            >>> def train_func():
+            ...     model = tf.keras.Sequential(
+            ...         [
+            ...             tf.keras.layers.InputLayer(input_shape=()),
+            ...             tf.keras.layers.Flatten(),
+            ...             tf.keras.layers.Dense(10),
+            ...             tf.keras.layers.Dense(1),
+            ...         ]
+            ...     )
+            ...     model.save("my_model.h5")
+            ...     checkpoint = TensorflowCheckpoint.from_h5("my_model.h5")
+            ...     session.report({"my_metric": 1}, checkpoint=checkpoint)
+
+            >>> trainer = TensorflowTrainer(
+            ...     train_loop_per_worker=train_func,
+            ...     scaling_config=ScalingConfig(num_workers=2))
+
+            >>> result_checkpoint = trainer.fit().checkpoint
+
+            >>> batch_predictor = BatchPredictor.from_checkpoint(
+            ...     result_checkpoint, TensorflowPredictor)
+            >>> batch_predictor.predict(ray.data.range(3))
+        """
+        assert path.isfile(file_path) and file_path.endswith(
+            ".h5"
+        ), "Please supply a h5 file path to `TensorflowCheckpoint.from_h5()`."
+        dir_path = path.dirname(os.path.abspath(file_path))
+        if preprocessor:
+            save_preprocessor_to_dir(preprocessor, dir_path)
```

Comment:
I remember discussing a bit on this before, but `from_h5` implies read only, yet we are writing a new file to a directory the user may not be expecting us to write to.

Should we create a new directory containing the preprocessor file and copy the h5 file there instead?

Reply:
I don't have a good answer here.
We can do that, but that would mean something like
```
with tempfile.TemporaryDirectory() as tmp_dir:
  shutil.copy(file_path, os.join(tmp_dir, file_name)
  save_preprocessor_to_dir(preprocessor, tmp_dir)
  checkpoint = Checkpoint.from_directory(tmp_dir)
  checkpoint_dict = checkpoint.to_dict()
  checkpoint2 = Checkpoint.from_dict(checkpoint_dict)
return checkpoint2
```

Alternatively, we could just mention in the doc string that one should expect some changes to the passed in directory. Most likely, this directory is the rank directory that we created for user to write stuff to anyways. 

I think this is something intrinsic to the current Checkpoint abstraction that is hard to work with.

Reply:
Hmm if we don't use a context manager then we can just return checkpoint instead of checkpoint2 right?

The rank directory is not really applicable if we are doing standalone batch prediction for example.

I think the tradeoff is if it's too expensive to do the `shutil.copy` and also how to clean up the directory that we created, is that right?

Reply:
If we dont use a context manager, then it is not clear whose responsibility it is to clean up this temporary folder right?
Since this is created in our code, it would be pretty reasonable to assume that we should do that. However, our Checkpoint class requires that this temporary directory to be valid even beyond the scope of this function. IMO there is not really any solution with a clear semantics. (and I don't like this. maybe we should bring this up in checkpoint PRD discussion)

So right now, I just added a few comments to the method informing the side effect. Let me know what you think.